docs/users/step-debugging.md

```diff
@@ -1,90 +1,90 @@
 
-<a name="phpstorm"></a>
+#### Visual Studio Code (vscode)
 
-### PhpStorm Debugging Setup
-
-[PhpStorm](https://www.jetbrains.com/phpstorm/download) is a leading PHP development IDE with extensive built-in debugging support. It provides two different ways to do debugging. One requires very little effort in the PhpStorm IDE (they call it zero-configuration debugging) and the other requires you to set up a "run configuration", and is basically identical to the Netbeans or Eclipse setup.
-
-**If you are using PhpStorm inside WSL2 (or perhaps other Linux configurations), under `Help→ Edit Custom VM Options`, add an additional line: `-Djava.net.preferIPv4Stack=true` This makes PhpStorm listen for Xdebug using IPV4; the Linux version of PhpStorm seems to default to using only IPV6.**
-
-#### PhpStorm Zero-Configuration Debugging
+1. Install the [php-debug](https://marketplace.visualstudio.com/items?itemName=felixfbecker.php-debug) extension.
+2. Update the project's [launch.json] (in `.vscode/launch.json`)  to add "Listen for xdebug" (
+   see [config snippet](snippets/launch.json)). For more on launch.json,
+   see [vscode docs](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations).
+3. Set a breakpoint in your index.php. If it isn't solid red, restart.
+4. In the menu, choose Run->Start Debugging.You may have to select "Listen for XDebug" by the green arrowhead at the top
+   left. The bottom pane of vscode should now be orange (live) and should say "Listen for XDebug".
 
-PhpStorm [zero-configuration debugging](https://confluence.jetbrains.com/display/PhpStorm/Zero-configuration+Web+Application+Debugging+with+Xdebug+and+PhpStorm) will automatically detect a connection and offer to create a "server", a file mapping from your workstation to the container. This means you only have to:
+Note that if you are using vscode on Windows with WSL2, you'll want the "PHP Debug" extension enabled in your distro (
+for example, Ubuntu). You'll also need the "Remote - WSL" extension enabled. vscode will suggest both of these to you if
+you have WSL2 enabled and a PHP project.
 
-1. Toggle the “Start Listening for PHP Debug Connections” button:
-  ![Start listening for debug connections button](images/phpstorm_listen_for_debug_connections.png)
-2. Set a breakpoint.
-3. Visit a page that should stop in the breakpoint you set.
-4. PhpStorm will ask you what mapping to use, so it can figure out how to map the path inside the container to the path on your workstation. The first time you do this with a project, PhpStorm will pop up a "server" dialog box to set the mapping. The default will work, but it's best to click the checkbox to map the whole project directory to /var/www/html.
+### Debug a website
 
-Note when using this recommended option:
+To start and stop Xdebug from the browser, you need to set a cookie parameter. You can use one of the
```

Comment:
You can start debugging sessions without any configuration now. If you like xdebug sessions to start automatically you have to run `ddev xdebug autostart`.

Reply:
That's not my experience. You have to `ddev debug autostart` to do it with no configuration.

Or maybe it's because all existing projects have `xdebug_enabled: false` in them?

Reply:
> Or maybe it's because all existing projects have xdebug_enabled: false in them?

Yes that's proabably the reason. I noted this in the manual testing instructions.
Of course we should not change the behavior for projects that already have `xdebug_enabled: false` set explicitly. Maybe this was done by intend.
For new project `xdebug_enabled` should not appear in the config.yaml at all when running `ddev config` and the default will be `on`. As you already noticed the "new on" won't have any negative effect.

Reply:
All projects in the world have `xdebug_enabled: false`, because it's been the default since the beginning (before `ddev xdebug` even existed, when you had to change this setting to debug).  So we have to figure our way around this bit of friction.